### PR TITLE
Feature/az auth

### PR DESCRIPTION
**Pull request by jeremiahwander**

### CODE REVIEW

File: cpg_utils/cloudpath_hail_az.py [hunk 1]
```diff
@@ -4,26 +4,90 @@
...
+import os
+import json
+import logging
+from typing import Union, Optional, Any, Callable
 from urllib.parse import urlparse
-
+import mimetypes
+from azure.identity import DefaultAzureCredential, ClientSecretCredential
+from azure.storage.blob import BlobServiceClient
+#import azure.storage.blob
```
Comment (gregsmi) [inline cpg_utils/cloudpath_hail_az.py:15]: vestigial?

Comment (jeremiahwander) [reply cpg_utils/cloudpath_hail_az.py:15]: the azure.* imports are used in `HailAzureBlobClient.__init__()`, `mimetypes` is only used to provide a default value for constructor's parameter `content_type_method` which gets passed to the superclass constructor.

File: cpg_utils/cloudpath_hail_az.py [hunk 2]
```diff
@@ -4,26 +4,90 @@
...
+            content_type_method (Optional[Callable]): Function to call to guess media type (mimetype) when
+                writing a file to the cloud. Defaults to `mimetypes.guess_type`. Must return a tuple (content type, content encoding).
+        """
+        if account_url is None:
+            raise ValueError("account_url must be specified")
+
+        parsed = urlparse(account_url)
+        if parsed.query is not None and len(parsed.query) > 0:
+            # Use passed embedded SAS token, not sure why the length check is necessary
+            print(f'using SAS token, query={parsed.query}')
```
Comment (gregsmi) [inline cpg_utils/cloudpath_hail_az.py]: you want the prints in the final output?

File: cpg_utils/cloudpath_hail_az.py [hunk 3]
```diff
@@ -4,26 +4,87 @@
...
+                https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobserviceclient?view=azure-python).
+            local_cache_dir (Optional[Union[str, os.PathLike]]): Path to directory to use as cache
+                for downloaded files. If None, will use a temporary directory.
+            content_type_method (Optional[Callable]): Function to call to guess media type (mimetype) when
+                writing a file to the cloud. Defaults to `mimetypes.guess_type`. Must return a tuple (content type, content encoding).
+        """
+        if account_url is None:
+            raise ValueError("account_url must be specified")
+
+        parsed = urlparse(account_url)
```
Comment (gregsmi) [inline cpg_utils/cloudpath_hail_az.py:55]: the result here seems to be a complicated named tuple, which may affect your truthiness check? might be easier to deal with it unpacked - i.e. _, _, _, _, query, _ = ...

**Review by gregsmi — APPROVED** [review]

**Pull request merged**